### PR TITLE
[DOP-7694] Add Connection.get_packages() method

### PR DESCRIPTION
**Pull request by dolfinus**
## Change Summary

* Added `Connection.get_packages()` method to all DB connection classes (+Kafka). Input arguments are different for different connections, but result is always `list[str]`, containing names and versions of Maven packages.
* Some connections have additional checks for input arguments, to improve user experience.
* Marked syntax `Connection.package` and `Connection.package_spark_X_Y` as deprecated.
* Updated examples to use new syntax instead of old one.
* Added tests covering these changes.

## Checklist

* [X] Commit message and PR title is comprehensive
* [X] Keep the change as small as possible
* [X] Unit and integration tests for the changes exist
* [X] Tests pass on CI and coverage does not decrease
* [X] Documentation reflects the changes where applicable
* [X] `docs/changelog/next_release/<pull request or issue id>.<change type>.rst` file added describing change
  (see [CONTRIBUTING.rst](https://github.com/MobileTeleSystems/onetl/blob/develop/CONTRIBUTING.rst) for details.)
* [X] My PR is ready to review.

### CODE REVIEW

File: onetl/_util/classproperty.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    ...        return 123
+    >>> # no call
+    >>> assert My.attribute == 123
+    """
+
+    def __init__(self, f):
+        self.f = f
+
+    def __get__(self, obj, objtype=None):
+        return self.f(objtype or type(obj))
```
Comment (dmitry-pedchenko) [inline onetl/_util/classproperty.py:37]: когда doctest тут вроде надо писать

```suggestion
        return self.f(objtype or type(obj))

if __name__ == "__main__":
    import doctest
    doctest.testmod()
```

Comment (dolfinus) [reply onetl/_util/classproperty.py:37]: Для запуска doctest я использую pytest --doctest-modules, см. pytest.ini

**Review by dmitry-pedchenko — APPROVED** [review]

File: onetl/connection/db_connection/clickhouse.py [hunk 2]
```diff
@@ -100,12 +104,15 @@ class Clickhouse(JDBCConnection):
 
         extra = {"continueBatchOnError": "false"}
 
+        # Create Spark session with Clickhouse driver loaded
+        maven_packages = Clickhouse.get_packages
```
Comment (andy-takker) [inline onetl/connection/db_connection/clickhouse.py:108]:
```suggestion
        maven_packages = Clickhouse.get_packages()
```

File: onetl/connection/db_connection/mongodb.py [hunk 3]
```diff
@@ -261,9 +261,81 @@ class Config:
...
+
+        if scala_ver.digits(2) < (2, 12) or scala_ver.digits(2) > (2, 13):
+            raise ValueError(f"Scala {scala_ver} is not supported by MongoDB connector")
+
+        return [f"org.mongodb.spark:mongo-spark-connector_{scala_ver.digits(2)}:10.1.1"]
+
+    @classproperty
+    def package_spark_3_2(cls) -> str:
+        """Get package name to be downloaded by Spark 3.2."""
+        msg = "`MongoDB.package_3_2` will be removed in 1.0.0, use `MongoDB.get_packages(spark_version='3.2')` instead"
```
Comment (andy-takker) [inline onetl/connection/db_connection/mongodb.py:322]:
```suggestion
        msg = "`MongoDB.package_spark_3_2` will be removed in 1.0.0, use `MongoDB.get_packages(spark_version='3.2')` instead"
```
тут и еще ниже

File: onetl/_util/classproperty.py (hunk 1 above)
Comment (dolfinus) [reply onetl/_util/classproperty.py:37]: Сейчас проверю, падает ли pytest при ошибках в doctest

**Review by andy-takker — APPROVED** [review]

File: onetl/_util/classproperty.py (hunk 1 above)
Comment (dolfinus) [reply onetl/_util/classproperty.py:37]: Оказалось, что `pytest -m` не выполняет doctest (даже если в `-m` передан `not marker`), в документации про это не слова. Добавил в CI явный запуск `pytest onetl/_util`

Comment (dolfinus) [reply onetl/_util/classproperty.py:37]: Вот как выглядит упавший doctest в CI:
https://github.com/MobileTeleSystems/onetl/actions/runs/5690032876/job/15422667340?pr=87

**Pull request merged**